contracts/examples/digital-cash/src/digital_cash.rs

```diff
@@ -68,15 +78,35 @@ pub trait DigitalCash {
             deposit.expiration_round >= self.blockchain().get_block_round(),
             "deposit expired"
         );
+
+        const HASH_DATA_BUFFER_LEN: usize = 1024;
+
+        let sig_len = signature.len();
         require!(
-            self.crypto().verify_ed25519(
-                &address.to_byte_array()[..],
-                &caller_address.to_byte_array()[..],
-                signature.to_boxed_bytes().as_slice()
-            ),
-            "invalid signature"
+            sig_len <= HASH_DATA_BUFFER_LEN,
+            "Attributes too long, cannot copy into static buffer"
```

Comment:
Misleading error message
```suggestion
            "Data too long, cannot copy into static buffer"
```